### PR TITLE
SF-1171 Update to ParatextData 9.2.4-alpha8, account for HexId

### PR DESCRIPTION
**Pull request by RaymondLuong3**
ParatextData started using HexId for Paratext project IDs. There are
several places where we could draw the line between code using a
string for PT project ID, and code using a HexId for PT project ID.
This commit converts to/from HexId when using a type from
ParatextData, and attempts to retain the string type for PT project
IDs in SF code.
Also the code for a Head registry request was move into Paratext Data
and this removes it from SF.

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/949)

### CODE REVIEW

**Review by irahopkinson — COMMENTED** [review]
Although this is approved (just one non-blocking comment that is probably just a typo), its a significant enough change that I'd like to merge it after we have released the DBL work to Live and then give time for this to be well tested on QA. I've made this a draft PR until then.

> Reviewed 9 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/949#-:-MTFJHEL2CLCnGB3XBIO:b-3bdfsy)* status: all files reviewed, 1 unresolved discussion (waiting on @RaymondLuong3)

___
*[test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs, line 1036 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/949#-MTFITItAhQWHvF68s1O:-MTFITItAhQWHvF68s1P:b-m0mwzn) ([raw file](https://github.com/sillsdev/web-xforge/blob/4829beedd7b1cbc785d52256779ad9cc6c5aadd7/test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs#L1036)):*
> ```
> HexId.FromStrSafe
> ```
Why is this used here? Everywhere else is `HexId.FromStr`.

**Review by RaymondLuong3 — COMMENTED** [review]
I think that would be best to wait until after we ship to merge this so that it can be thoroughly tested.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/949#-:-MTGrHm18O6fAgfRRSvB:b-p97psv)* status: 8 of 9 files reviewed, 1 unresolved discussion (waiting on @irahopkinson)

___
*[test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs, line 1036 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/949#-MTFITItAhQWHvF68s1O:-MTGqoIk1XF2M2UyBqwJ:b-5bx2yc) ([raw file](https://github.com/sillsdev/web-xforge/blob/4829beedd7b1cbc785d52256779ad9cc6c5aadd7/test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs#L1036)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> HexId.FromStrSafe
> ```
Why is this used here? Everywhere else is `HexId.FromStr`.
</blockquote></details>

Yes, I had the same question too since I picked this up from Mark's branch. It may be that it was helpful in a test since it will never throw an error, but I think it would be best to have the error get thrown if HexId cannot parse projectId.

**Review by irahopkinson — APPROVED** [review]
:lgtm: 

> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/949#-:-MTeH2k94Qx4sJt2LBtz:bg7igkv)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by irahopkinson — APPROVED** [review]
> Reviewed 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/949)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**
